projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -72,332 +72,299 @@ const PRIORITIZED_INSERTER_BLOCKS = [ ...map( validFields, block => `jetpack/${
 const CUSTOMIZING_FORMS_URL = 'https://jetpack.com/support/jetpack-blocks/contact-form/';
 
-export const JetpackContactFormEdit = forwardRef(
-	(
-		{
-			attributes,
-			setAttributes,
-			postAuthorEmail,
-			hasInnerBlocks,
-			replaceInnerBlocks,
-			selectBlock,
-			clientId,
-			instanceId,
-			className,
-			blockType,
-			variations,
-			defaultVariation,
-			canUserInstallPlugins,
-			style,
-		},
-		ref
-	) => {
-		const {
-			to,
-			subject,
-			customThankyou,
-			customThankyouHeading,
-			customThankyouMessage,
-			customThankyouRedirect,
-			jetpackCRM,
-			salesforceData,
-		} = attributes;
-		const [ isPatternsModalOpen, setIsPatternsModalOpen ] = useState( false );
-
-		const blockProps = useBlockProps();
-		const { isLoadingModules, isChangingStatus, isModuleActive, changeStatus } =
-			useModuleStatus( 'contact-form' );
+function JetpackContactFormEdit( { name, attributes, setAttributes, clientId, className } ) {
+	const {
+		to,
+		subject,
+		customThankyou,
+		customThankyouHeading,
+		customThankyouMessage,
+		customThankyouRedirect,
+		jetpackCRM,
+		salesforceData,
+	} = attributes;
+	const instanceId = useInstanceId( JetpackContactFormEdit );
+	const { replaceInnerBlocks, selectBlock } = useDispatch( 'core/block-editor' );
+	const {
+		blockType,
+		canUserInstallPlugins,
+		defaultVariation,
+		variations,
+		hasInnerBlocks,
+		postAuthorEmail,
+	} = useSelect(
+		select => {
+			const { getBlockType, getBlockVariations, getDefaultBlockVariation } =
+				select( 'core/blocks' );
+			const { getBlocks } = select( 'core/block-editor' );
+			const { getEditedPostAttribute } = select( 'core/editor' );
+			const { getUser, canUser } = select( 'core' );
+			const innerBlocks = getBlocks( clientId );
 
-		const formClassnames = clsx( className, 'jetpack-contact-form', {
-			'is-placeholder': ! hasInnerBlocks && registerBlockVariation,
-		} );
+			const authorId = getEditedPostAttribute( 'author' );
+			const authorEmail = authorId && getUser( authorId ) && getUser( authorId ).email;
+			const submitButton = innerBlocks.find( block => block.name === 'jetpack/button' );
+			if ( submitButton && ! submitButton.attributes.lock ) {
+				const lock = { move: false, remove: true };
+				submitButton.attributes.lock = lock;
+			}
 
-		const isSalesForceExtensionEnabled =
-			!! window?.Jetpack_Editor_Initial_State?.available_blocks[
-				'contact-form/salesforce-lead-form'
-			];
+			return {
+				blockType: getBlockType && getBlockType( name ),
+				canUserInstallPlugins: canUser( 'create', 'plugins' ),
+				defaultVariation: getDefaultBlockVariation && getDefaultBlockVariation( name, 'block' ),
+				variations: getBlockVariations && getBlockVariations( name, 'block' ),
+				hasInnerBlocks: innerBlocks.length > 0,
+				postAuthorEmail: authorEmail,
+			};
+		},
+		[ clientId, name ]
+	);
+	const [ isPatternsModalOpen, setIsPatternsModalOpen ] = useState( false );
+	const wrapperRef = useRef();
+	const innerRef = useRef();
```

Comment:
The `wrapperRef` is for `ThemeProvider` and its `targetDom` prop.

The `innerRef` is replacing the `fowardRef` that wrapped the component and was used for `withStyleVariables`. Now that the logic from that HOC is moved into the component, `forwardRef` isn't needed and an internal ref can be used instead.